### PR TITLE
feat(webhooks): link shipment to order

### PR DESCRIPTION
**Pull request by joerivanveen**
Allows showing shipment details including barcode in order mode, when label is printed in backoffice.

INT-627

### CODE REVIEW

File: tests/Unit/App/Webhook/Hook/ShipmentStatusChangeWebhookTest.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            'shipment_id' => 192031595,
+            'account_id' => 162450,
+            'order_id' => 'api-uuid-string',
+            'shop_id' => 83287,
+            'status' => 2,
+            'barcode' => '3SHOHR763563926',
+            'shipment_reference_identifier' => '',
+        ],
+    ],
+]);
```
Comment (FreekVR) [inline tests/Unit/App/Webhook/Hook/ShipmentStatusChangeWebhookTest.php:95]: Please restore empty newline end-of-file

File: tests/Unit/App/Order/Repository/AbstractPdkOrderRepositoryTest.php [hunk 2]
```diff
@@ -44,3 +46,32 @@
...
+                    'level'   => 'notice',
+                    'message' => '[PDK]: Implement getByApiIdentifier, in PDK v3 it will be required.',
+                    'context' =>
+                        [
+                            'class'   => 'MyParcelNL\\Pdk\\App\\Order\\Repository\\AbstractPdkOrderRepository',
+                        ],
+                ],
+            ]
+        );
+});
```
Comment (FreekVR) [inline tests/Unit/App/Order/Repository/AbstractPdkOrderRepositoryTest.php:77]: Please restore empty newline end-of-file

**Review by FreekVR — CHANGES_REQUESTED** [review]

**Comment by FreekVR** [comment]
Commit message covers the end-user feature but not really the changes in this repo.

Suggest to change it to (as I understand these changes, please correct if not right)

```
feat: improved linking shipments to orders

allows showing barcode when printed from backoffice #327
```

File: src/App/Action/Backend/Shipment/UpdateShipmentsAction.php [hunk 3]
```diff
@@ -55,6 +55,10 @@ public function handle(Request $request): Response
         $orders    = $this->pdkOrderRepository->getMany($this->getOrderIds($request));
         $shipments = $this->shipmentRepository->getShipments($this->getShipmentIds($request, $orders));
 
+        if ($request->get('linkFirstShipmentToFirstOrder')) {
```
[the remark this answers is not in the document]
Comment (joerivanveen) [reply src/App/Action/Backend/Shipment/UpdateShipmentsAction.php]: Done.

File: src/App/Order/Repository/AbstractPdkOrderRepository.php [hunk 4]
```diff
@@ -19,6 +20,19 @@ abstract class AbstractPdkOrderRepository extends Repository implements PdkOrder
      */
     abstract public function get($input): PdkOrder;
 
+    // TODO: v3.0.0 make method abstract to force implementation
+    public function getByApiIdentifier(string $uuid): ?PdkOrder
```
[the remark this answers is not in the document]
Comment (joerivanveen) [reply src/App/Order/Repository/AbstractPdkOrderRepository.php:30]: Done.

File: src/App/Webhook/Hook/ShipmentStatusChangeWebhook.php [hunk 5]
```diff
@@ -20,9 +22,16 @@ public function handle(Request $request): void
     {
         $content = $this->getHookBody($request);
 
+        // translate order_id (which is api identifier / uuid) to local order id for db
+        $order = Pdk::get(PdkOrderRepositoryInterface::class)->getByApiIdentifier($content['order_id']);
```
[the remark this answers is not in the document]
Comment (joerivanveen) [reply src/App/Webhook/Hook/ShipmentStatusChangeWebhook.php:26]: Done.

**Review by FreekVR — APPROVED** [review]

**Review by GravendeelJochem — APPROVED** [review]

**Pull request merged**